File: src/main/java/common/item/ItemStack.java
package common.item;

import common.registries.Blocks;
import common.registries.Items;

public class ItemStack {
    private Item item;
    private int amount;

    public ItemStack(Item item, int amount) {
        this.item = item;
        this.amount = amount;
    }

    public Item getItem() {
        return item;
    }

    public void setItem(Item item) {
        this.item = item;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
        if (this.amount==0){
            setItem(Items.getItem("air"));
            this.amount=-1;
        }
    }
}

File: src/main/java/common/item/ItemStack.kt
package common.item

import common.registries.Items

// TODO: This class should have a limit of sorts as a static variable.
class ItemStack(var item: Item, private var amount: Int) {

    fun getAmount(): Int {
        return amount
    }

    fun setAmount(amount: Int) {
        this.amount = amount
        if (this.amount == 0) {
            item = Items.getItem("air")
            this.amount = -1
        }
    }
}
